### PR TITLE
Next JS migration

### PR DESCRIPTION
**Pull request by epicfaace**
## Reasons for making this change

Faster on the client side to use SSR pages. Also, then we can deploy to AWS using the same configuration we're using for stanforddaily.com

Run `yarn dev` to start, and `yarn deploy` to deploy.

Deployed at: https://d148m2cwwi25lc.cloudfront.net/

### CODE REVIEW

**Comment by ufxela** [comment]
I think the calendar is broken e.g. if you go to http://localhost:3000/1895/01/ then I get an unhandled runtime error

File: components/AllYearView.jsx [hunk 1]
```diff
@@ -41,13 +39,13 @@ class AllYearView extends React.Component {
         let dateMoment = moment({ year: year, month: month });
         monthsElements.push(
           <div key={month} className="EachMonth">
-            {isMonthInMetaData(this.state.allPapers, dateMoment) ? (
+            {isMonthInMetaData(allPapers, dateMoment) ? (
               <Link
-                to={getMonthPath(dateMoment)}
+                href={getMonthPath(dateMoment)}
                 title={dateMoment.format("MMMM YYYY")}
```
Comment (ufxela) [inline components/AllYearView.jsx]: this is unrelated to your changes, i think, but I get a `Warning: Failed prop type: Link: unknown props found: title, className` warning here

**Review by ufxela — COMMENTED** [review]
I think the calendar is broken e.g. if you go to http://localhost:3000/1895/01/ then I get an unhandled runtime error

**Comment by epicfaace** [comment]
> I think the calendar is broken e.g. if you go to http://localhost:3000/1895/01/ then I get an unhandled runtime error

@ufxela hmm it's working for me -- can you try doing `yarn` and try again?

image

**Review by ufxela — APPROVED** [review]
calendar works now + no more error in console

**Comment by epicfaace** [comment]
We still have the issue where going directly to https://d148m2cwwi25lc.cloudfront.net/calendar and https://d148m2cwwi25lc.cloudfront.net/acknowledgements doesn't work. No idea why this is happening

**Comment by ufxela** [comment]
> We still have the issue where going directly to https://d148m2cwwi25lc.cloudfront.net/calendar and https://d148m2cwwi25lc.cloudfront.net/acknowledgements doesn't work. No idea why this is happening

huh weird. also https://d148m2cwwi25lc.cloudfront.net/acknowledgements works for me. Calendar is broken though.

**Comment by epicfaace** [comment]
> also https://d148m2cwwi25lc.cloudfront.net/acknowledgements works for me.

working for me too now!

**Comment by ufxela** [comment]
oddly, even though month pages don't work on cloudfront, they do work when running locally. Huh.

I do get this warning when running locally and going to a month page

```
Warning: componentWillMount has been renamed, and is not recommended for use. See https://fb.me/react-unsafe-component-lifecycles for details.

* Move code from componentWillMount to componentDidMount (preferred in most cases) or the constructor.

Please update the following components: Uncontrolled(Calendar)
    in Uncontrolled(Calendar)
    in ForwardRef (at CalendarView.jsx:96)
    in div (at CalendarView.jsx:95)
    in div (at CalendarView.jsx:74)
    in CalendarView
    in withRouter(CalendarView) (at _app.jsx:46)
    in div (at _app.jsx:45)
    in div
    in Div100vh (at _app.jsx:33)
    in App
    in Unknown
    in Context.Provider
    in Context.Provider
    in Context.Provider
    in AppContainer
```

**Comment by ufxela** [comment]
updated big calendar to newer version, so I no longer get that warning. Maybe that will fix cloudfront? idk

**Pull request merged**